src/cluster.c

```diff
@@ -1876,7 +1953,7 @@ int clusterHandshakeInProgress(char *ip, int port, int cport) {
 
         if (!nodeInHandshake(node)) continue;
         if (!strcasecmp(node->ip,ip) &&
-            node->port == port &&
+            (node->tcp_port == port || node->tls_port == port) &&
```

Comment:
```suggestion
            getNodeDefaultClientPort(node) &&
```
For compatibility, we have to assume that the port + IP we provided in cluster meet, or provided by another node, follows our normal logic about server.cluster_tls. If someone provides both, I don't think we should deduplicate them here, because one of them will fail.